src/main/java/dota/buff/service/impl/MatchServiceImpl.java

```diff
@@ -7,27 +7,36 @@
 import dota.buff.model.HeroDTO;
 import dota.buff.model.MatchDTO;
 import dota.buff.model.PlayerDTO;
+import dota.buff.model.enums.Side;
 import dota.buff.service.HeroService;
 import dota.buff.service.MatchService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
 
+@Service
 public class MatchServiceImpl implements MatchService {
 
-    private static final String API_KEY = "B946D0A6B0726327F1C9D44095CD11C0";
-    private static final Dota2ApiClient client = new Dota2ApiClient(API_KEY);
-    private static final HeroService heroService = new HeroServiceImpl();
+    private final Dota2ApiClient client;
+    private final HeroService heroService;
+
+    @Autowired
+    public MatchServiceImpl(Dota2ApiClient client, HeroService heroService) {
+        this.client = client;
+        this.heroService = heroService;
+    }
 
     @Override
     public MatchDTO getMatchById(long matchId) {
         MatchDetail matchDetail = client.send(getMatchDetailRequest(matchId));
         return new MatchDTO(
                 matchDetail.getMatchId(),
                 matchDetail.getGameMode().toString(),
-                convertAllMatchPlayers(matchDetail),
-                convertAllMatchHeroes(convertAllMatchPlayers(matchDetail)),
-                matchDetail.isRadiantWin(),
+                convertAllMatchPlayers(matchDetail.getPlayers()),
+                convertAllMatchHeroes(convertAllMatchPlayers(matchDetail.getPlayers())),
+                getWinner(matchDetail),
```

Comment:
Introduce a local var for convertAllMatchPlayers plz
` List<PlayerDTO> allMatchPlayers = convertAllMatchPlayers(matchDetail.getPlayers());`
In order to avoid calling this method twicely

Reply:
What ?

Reply:
Done